pkg/athena/api/api.go

```diff
@@ -2,34 +2,41 @@ package api
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/service/athena"
 	"github.com/aws/aws-sdk-go/service/athena/athenaiface"
 	"github.com/grafana/athena-datasource/pkg/athena/models"
 	"github.com/grafana/grafana-aws-sdk/pkg/awsds"
+	"github.com/grafana/grafana-aws-sdk/pkg/sql/api"
+	"github.com/grafana/sqlds/v2"
 )
 
 type API struct {
 	Client   athenaiface.AthenaAPI
 	settings *models.AthenaDataSourceSettings
 }
 
-func New(sessionCache *awsds.SessionCache, settings *models.AthenaDataSourceSettings) (*API, error) {
-	region := settings.DefaultRegion
-	if settings.Region != "" {
-		region = settings.Region
-	}
-	session, err := sessionCache.GetSession(region, settings.AWSDatasourceSettings)
+func New(sessionCache *awsds.SessionCache, settings *models.AthenaDataSourceSettings) (api.AWSAPI, error) {
+	sess, err := awsds.GetSessionWithDefaultRegion(sessionCache, settings.AWSDatasourceSettings)
 	if err != nil {
 		return nil, err
 	}
-	return &API{athena.New(session), settings}, nil
+
+	svc := athena.New(sess)
+	svc.Handlers.Send.PushFront(func(r *request.Request) {
+		r.HTTPRequest.Header.Set("User-Agent", awsds.GetUserAgentString("Athena"))
+	})
+
+	return &API{svc, settings}, nil
 }
 
-func (c *API) Execute(ctx context.Context, query string) (*athena.StartQueryExecutionOutput, error) {
-	input := &athena.StartQueryExecutionInput{
-		QueryString: aws.String(query),
+func (c *API) Execute(ctx context.Context, input *api.ExecuteQueryInput) (*api.ExecuteQueryOutput, error) {
```

Comment:
Really like the abstractions you introduced by moving the SQL and Resource interfaces to the sdk. Might be just me though, but while implementing the SQL interface here (with the resources interface) the method names seem a bit confusing. Any chance we can find names that provides some context, like `ExecuteQuery`, `GetQueryStatus`. Thoughts?

Reply:
I think that adding `Query` to the function name is redundant? (since it's already a parameter) For example, the SQL driver defines [`sql.Exec(query string)`](https://pkg.go.dev/database/sql#DB.Exec) (not `sql.ExecQuery(query string)`). It's a bit more indirect in this case but still the same logic applies.